### ISSUE TITLE
Sort column overrides existing columns

### ISSUE THREAD

**Issue by LiterallyUniqueLogin** [pola-rs/polars#2413]
#### Are you using Python or Rust?

**Python**.

#### What version of polars are you using?

0.12.13

#### What operating system are you using polars on?

CentOS Linux release 8.1.1911 (Core)

#### Describe your bug.

Sorting by an expression overwrites column in the dataframe with the same name

#### What are the steps to reproduce the behavior?

```
pl.DataFrame({"a": [1, 2, 1, 1], "c": [0, 1, 2, 3]}).with_column((pl.col('a') - pl.col('c')).alias('diff')).sort(pl.col('diff').abs())
shape: (4, 3)
┌─────┬─────┬──────┐
│ a   ┆ c   ┆ diff │
│ --- ┆ --- ┆ ---  │
│ i64 ┆ i64 ┆ i64  │
╞═════╪═════╪══════╡
│ 1   ┆ 0   ┆ 1    │
├╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌┤
│ 2   ┆ 1   ┆ 1    │
├╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌┤
│ 1   ┆ 2   ┆ 1    │
├╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌┤
│ 1   ┆ 3   ┆ 2    │
└─────┴─────┴──────┘
```
Notice how `diff` is in absolute value even though that wasn't the column added to the dataframe (only the temporary sort column)

**Comment by zundertj** [comment]
It probably overwrites it. Even smaller example:

```python
import polars as pl
df = pl.DataFrame({"a": [1, 2, -1, -2]})
df.sort(pl.col("a").abs())

```

```shape: (4, 1)
┌─────┐
│ a   │
│ --- │
│ i64 │
╞═════╡
│ 1   │
├╌╌╌╌╌┤
│ 1   │
├╌╌╌╌╌┤
│ 2   │
├╌╌╌╌╌┤
│ 2   │
└─────┘
```

The temporary workaround is create the column first, but it is very ugly (you can't alias the column inside the `sort` expression, which makes sense, otherwise that would be cleaner):
```python
df.with_column(pl.col("a").abs().alias("b")).sort(pl.col("b")).drop("b")
```

Python code uses LazyFrame, and invokes Rust as follows:
```python
self._ldf.sort_by_exprs(by, reverse)
```

**Comment by ritchie46** [comment]
Thanks @LiterallyUniqueLogin and @zundertj  for exactly pinpointing the issue. This indeeds overwrites the column where we should only use it for sorting. I will fix this.

**Comment by ritchie46** [comment]
Fixed in #2419
